src/hooks/authentication/useAuthToken/index.tsx

```diff
@@ -0,0 +1,42 @@
+import { useContext, useState } from 'react';
+import UserContext from 'pages/Login/UserContext';
+
+const getUserDetailsFromToken = (token: string) => {
+  const base64Url = token.split('.')[1];
+  const base64 = base64Url.replace('-', '+').replace('_', '/');
+  return JSON.parse(window.atob(base64));
+};
+
+const useAuthToken = () => {
+  const [authToken, setAuthTokenInternally] = useState<string | null>(() => {
+    return localStorage.getItem('token');
+  });
+  const { setUser } = useContext(UserContext);
+
+  const setAuthToken = (newToken: string | null) => {
+    if (!newToken) {
+      localStorage.removeItem('token');
+      localStorage.removeItem('username');
+      setUser({
+        username: '',
+        token: '',
+        isAuthenticated: false,
+      });
+      return;
+    }
+    localStorage.setItem('token', newToken as string);
```

Comment:
Does it need an expiration time?

Reply:
@mberacochea 
the backend sets the expiry to around 3 hours by default. 
I've written this to work in a way that, anytime a user tries to visit an auth protected route, a call is made to the /utils/token/verify endpoint. If the token is expired at that point, it will get removed from local storage, and the user will be logged out.